Svelte - case study

pretzL opened the pull request:
This case study should be more or less complete, hence why I am submitting it for review. I still need to write a summary and run the study through a spell checker or three, but the content is laid out according to what I had in mind.

Fermain commented on svelte-case-study.md:
```diff
@@ -0,0 +1,203 @@
+
+The key features of Svelte are:
+
+- No virtual DOM
```
Consider using a markdown table here since you have one heading and one subpoint each

pretzL replied:
I tried setting up a table, but I don't really feel like it's more organized or looks better. If I had more subpoints, it might fit better in my mind. I think I'll leave it as-is.

Fermain commented on svelte-case-study.md:
```diff
@@ -0,0 +1,203 @@
+
+When the value of visible changes, Svelte automatically updates the DOM to reflect the change and render or destroy the component as needed. This allows for easy control over component loading and unloading, and ensures that resources are only used when they are actually required, instead of constantly rendering every component like in some high-budget slideshow.
+
+<p align="center">
```
```suggestion
```

Fermain commented on svelte-case-study.md:
```diff
@@ -0,0 +1,203 @@
+
+<p align="center">
+  <img src="https://user-images.githubusercontent.com/30121693/217382445-d5addc54-21a2-48bd-b613-dd9cac2e657b.png" alt="Svelte's key features"/>
+</p>
```
```suggestion
```

Where possible we try to use as little HTML and handle formatting issues with CSS using consistent rules

Fermain commented on svelte-case-study.md:
```diff
@@ -0,0 +1,203 @@
+When the value of visible changes, Svelte automatically updates the DOM to reflect the change and render or destroy the component as needed. This allows for easy control over component loading and unloading, and ensures that resources are only used when they are actually required, instead of constantly rendering every component like in some high-budget slideshow.
+
+<p align="center">
+  <img src="https://user-images.githubusercontent.com/30121693/217382445-d5addc54-21a2-48bd-b613-dd9cac2e657b.png" alt="Svelte's key features"/>
```
```suggestion
![Svelte's key features](https://user-images.githubusercontent.com/30121693/217382445-d5addc54-21a2-48bd-b613-dd9cac2e657b.png)
```

Fermain reviewed (commented):
This is a well written and engaging case study that keeps the reader amused throughout, while not straying too far away from the technical tone expected.

There are some formatting changes to be made, mostly changing the syntax for images as this is handled by the system.

It might be nice to mention Sveltekit somewhere even as passing for those who want to explore full stack options.

sabaFitwi commented on svelte-case-study.md:
```diff
@@ -0,0 +1,203 @@
+
+Although Svelte, React, Vue.js and Angular are all popular JavaScript frameworks for building web applications, they have some very different design philosophies, strengths and weaknesses depending on their use.
+
+Svelte has a relatively simple syntax and a different approach to building web applications, making it relatively easy to learn. React has a steeper learning curve due to its use of a virtual DOM, although it has a large and supportive community. Vue.js has a similar learning curve to React, but with a simpler syntax. Angular has the steepest learning curve of the four frameworks, due to its comp ...
```
is it **whereas** or where as? i am not 100% sure  but check also if medium**-** to large-size or simply medium to large-size****

pretzL replied:
![Screenshot_20230214-150918](https://user-images.githubusercontent.com/30121693/218762628-fec3911c-de85-4791-82b8-93417e17dc83.png)
Good catch, thanks

WParsec commented:
This was both fun and well written. You have almost managed to shift my interest from React to Svelte which I deem as a successful case study. I am left with but one question after reading it; 
Svelte in context of job opportunities and industry usage. 
You touch on the subject, but if I am to at all criticise, I would like more. 
Well done!

pretzL commented:
> This was both fun and well written. You have almost managed to shift my interest from React to Svelte, which I deem as a successful case study. I am left with but one question after reading it; Svelte in context of job opportunities and industry usage compared to other frameworks. You touch on the subject, but if I am to at all criticise, I would like more. Well done!

Thank you!
The problem is finding accurate data on how widespread Svelte actually is in the industry. I can tell you now we're using it at work and will continue to use it in major projects. I predict Svelte will likely be the most used framework within the next 2 years, but my prediction isn't suited for a technical report.

NehGuk commented:
Hey! The article reads well and engaging. Here's my peer-feedback:

> Svelte uses a reactive and declarative syntax for building UI components, making it easy to understand and use for developers of all skill levels.

Not sure who exactly your target audience is, but I would assume that not everyone knows what "reactive and declarative syntax" is. Sure, even amongst second year students. I would include a short sentence to briefly explain it upon the first occurrence of the term.

> ... performing specific tasks or adding specific functionality to your apps.

Does it sound better if we use both objects in their plural form? "Tasks" and "functionalities"? Or maybe it's just my OCD...

> applications, where as React and Vue are...

Don't take my word, but isn't it "whereas" instead of "where as"?

> ... and retention, but lacks behind in usage.

I believe you meant "lags behind" instead of "lacks behind", right?

> Passing props

Similarly to a previous comment, students who do not have experience with frameworks do not know what "props" are. Why not explain what it is?

> starting to favor it

Following the Noroff style guide, should we use "favour" instead of "favor"? I'm not sure myself.

> Richard Dawkins

Why on earth do we have "Richard Dawkins" at the end as a reference? Do you expect that every reader know that Richard Dawkins sort of invented the concept of "meme", as an analogy to the concept of "gene", and that connects to the fact that you used a lot of "memes" in your article? hehehe... I like the joke. But, personaly, I diskike when authors write things that convey the message "hey, look how smart I am, I can even write stuff that you may not understand", when they really could write things that convey the message "hey, look, if this is new to you, let me explain how it works". There's that famous quote wrongly attributed to Albert Einstein, that goes like this: "You do not really understand something unless you can explain it to your grandmother." Oh... My apologies for the digression :)

As an intro to Svelte, I think the article works really well for students who already have some knowledge on javascript frameworks. Not so much as an introduction to the topic. 

If (A) your target audience is people who already know a bit about JS frameworks, then the article is great. If (B) your target audience is students who have not yet been introduced to JS frameworks, I believe the approach should be different. I'm considering that you chose A, in which case, great job! :) I really liked the article (and the memes).

pretzL commented:
> > Svelte uses a reactive and declarative syntax for building UI components, making it easy to understand and use for developers of all skill levels.
> Not sure who exactly your target audience is, but I would assume that not everyone knows what "reactive and declarative syntax" is. Sure, even amongst second year students. I would include a short sentence to briefly explain it upon the first occurrence of the term.

I've added an explanation in the section below the bullet points. My target audience is people approximately at the skill level of our current course progression, so "has a vague idea of what a JavaScript framework is".

> > ... performing specific tasks or adding specific functionality to your apps.
> 
> Does it sound better if we use both objects in their plural form? "Tasks" and "functionalities"? Or maybe it's just my OCD...

I agree, so I've changed it to "functionalities".

> > applications, where as React and Vue are...
> 
> Don't take my word, but isn't it "whereas" instead of "where as"?

You're correct. Saba also commented on this before, but for some reason that change didn't seem to go through. It should've now though.

> > ... and retention, but lacks behind in usage.
> 
> I believe you meant "lags behind" instead of "lacks behind", right?

Correct, fixed!

> > Passing props
> 
> Similarly to a previous comment, students who do not have experience with frameworks do not know what "props" are. Why not explain what it is?

Added a section to explain this.

> > starting to favor it
> 
> Following the Noroff style guide, should we use "favour" instead of "favor"? I'm not sure myself.

Although it pains me, it has been done.

> > Richard Dawkins
> 
> Why on earth do we have "Richard Dawkins" at the end as a reference? Do you expect every reader to know that Richard Dawkins sort of invented the concept of "meme", as an analogy to the concept of "gene", and that connects to the fact that you used a lot of "memes" in your article? hehehe... I like the joke. But, personaly, I diskike when authors write things that convey the message "hey, look how smart I am, I can even write stuff that you may not understand", when they really could write things that convey the message "hey, look, if this is new to you, let me explain how it works". There's that famous quote wrongly attributed to Albert Einstein, that goes like this: "You do not really understand something unless you can explain it to your grandmother." Oh... My apologies for the digression :)

I'm gonna try to comment on this as nicely as I can because you come off as quite rude. I don't see where I fit your notion, unless you consider failing to explain three keywords to equal "hey, look how smart I am, I can even write stuff that you may not understand", but I'll try to look at this as some weird form of constructive criticism regarding explaining certain concepts more in-depth to fit a broader audience. If your comment was in regards to the Richard Dawkins credit being some kind of convoluted joke, see below.

This is the reason for the Richard Dawkins credit:
![image](https://user-images.githubusercontent.com/30121693/219583908-61cd058f-4072-4fff-b81c-4f6d60e1f204.png)

> If (A) your target audience is people who already know a bit about JS frameworks, then the article is great. If (B) your target audience is students who have not yet been introduced to JS frameworks, I believe the approach should be different. I'm considering that you chose A, in which case, great job! :) I really liked the article (and the memes).

A bit of both, really. It's meant for students who vaguely know of JavaScript frameworks but haven't necessarily done anything in any yet. Svelte is a great starter framework because of how simple it is to use. I do, however, agree that some concepts could've been explained more in-depth.

Thanks for your in-depth review and feedback, even if the last paragraph leaves a bit of a sour taste.

NehGuk commented:
Cool! Btw, I really apologise if you felt that the last paragraph sounded "sour". That was **absolutely not the intended mood**, although I now see why it may have felt otherwise. It was just a light-hearted criticism. As I said, I do like the Dawkins joke, but this reference is buried under several layers of non-explicit knowledge. I think it shouldn't be a reference in a report like that. But that does raise a strange question: how on earth do we correctly reference memes from the internet? I have no idea, hehehe :)

Fermain commented:
Honestly, linking to know your meme might be the best way forwards

Fermain reviewed (approved):
A witty, enjoyable case study on a highly relevant technology!
